src/Microsoft.AspNetCore.Hosting/WebHostBuilder.cs

```diff
@@ -185,23 +240,29 @@ private IServiceCollection BuildCommonServices(out AggregateException hostingSta
             var appEnvironment = PlatformServices.Default.Application;
             var contentRootPath = ResolveContentRootPath(_options.ContentRootPath, appEnvironment.ApplicationBasePath);
             var applicationName = _options.ApplicationName ?? appEnvironment.ApplicationName;
+            var hostingContext = new HostingContext();
 
             // Initialize the hosting environment
             _hostingEnvironment.Initialize(applicationName, contentRootPath, _options);
+            hostingContext.HostingEnvironment = _hostingEnvironment;
```

Comment:
amusingly the hosting environment is built from config, before we use it to build more config. That's going to be really confusing if you want to change the config source the environment builds from.